### DISCUSSION TITLE
JIT compile issue when using RNN as a branch

### DISCUSSION THREAD

**Discussion by ghost** [lululxvi/deepxde#1438] [Q&A]
Hi, I wish to resolve an issue with JIT compile when using our self-defined RNN network as a branch.

We used RNN models in the branch network, defining the branch with tf.keras.models.Sequential to build RNN models such as GRU or LSTM.

Each time, we had to comment out every line in the model.py (under .local/lib/python3.10/site-packages/deepxde) file such as below.

> #@tf.function(jit_compile=config.xla_jit)

Otherwise, the code was stuck before the training process and would never proceed.
If commenting, the code ran successfully on CPU, but not on GPU.

It would be wonderful to know what may cause the problem and how to make it run on GPU.

```data = dde.data.TripleCartesianProd(x_train, y_train, x_test, y_test)

activation = "relu"

branch = tf.keras.models.Sequential([
     tf.keras.layers.GRU(units=256,batch_input_shape=(batch_size, m, 1),activation = 'tanh',return_sequences = True, dropout=0.00, recurrent_dropout=0.00),
     tf.keras.layers.GRU(units=128,activation = 'tanh',return_sequences = False, dropout=0.00, recurrent_dropout=0.00),
     tf.keras.layers.RepeatVector(m),
     tf.keras.layers.GRU(units=128,activation = 'tanh',return_sequences = True, dropout=0.00, recurrent_dropout=0.00),
     tf.keras.layers.GRU(units=256,activation='tanh',return_sequences = True, dropout=0.00, recurrent_dropout=0.00),
     tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(1)),
     tf.keras.layers.Flatten(),])

branch.summary()
    
net = dde.maps.DeepONetCartesianProd(
        [m, branch], [2, 101, 101, 101, 101, 101, 101], activation, "Glorot normal")


model = dde.Model(data, net)
model.compile(
    "adam",
    lr=1e-3,
    decay=("inverse time", 1, 1e-4),
    metrics=["mean l2 relative error"],
)
```

**Comment by lululxvi** [comment]
You can use `dde.config.disable_xla_jit()` https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.config.disable_xla_jit

**Reply by ghost** [reply]
Thank you so much for the reply, Dr. Lu Lu.
One issue is that if we disable jit compile with dde.config.disable_xla_jit(), we are unable to utilize the whole GPU performance and it is too slow when training.
Is there a way to use jit and run RNNs (GRU, LSTM) ?

**Reply by lululxvi** [reply]
I don't know. This is a problem of TensorFlow.

**Reply by ghost** [reply]
I tried dde.config.disable_xla_jit() with the DeepXDE version 1.9.3, and it did help! The training was way faster (3 times faster for the RNN branch case in my practice) than commenting out all the jit lines in model.py file. Thank you Dr. Lu Lu for the help. I really appreciate it.
